sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -19,6 +21,127 @@
 from ...utils._encode import _unique
 
 
+def get_categorical_colors_and_mapping(value_list, palette="tab10"):
+    """
+    Derives a list of color and color mapping from the passed value list.
+
+    Parameters
+    ----------
+    value_list : array-like of shape (n_samples,) of str
+        Names of the categorical values by which the ICE lines are colored
+    palette: str
+        name of the matplotlib color palette
+
+    Returns
+    -------
+    list of colors, dictionary where the value are mapped to a color
+    """
+    color_palette = [
+        mpl.colors.to_hex(rgba_color)
+        for rgba_color in mpl.cm.get_cmap(
+            name=palette, lut=len(np.unique(value_list))
+        ).colors
+    ]
+    mapping = {v: color_palette[i] for i, v in enumerate(np.unique(value_list))}
+    color_list = [mapping[value_list[i]] for i in range(len(value_list))]
+    return color_list, mapping
+
+
+def get_continuous_colors_and_cmap_norm(value_list, palette="viridis"):
+    """
+
+
+    Parameters
+    ----------
+    value_list : array-like of shape (n_samples,) of int or float
+        Numerical values by which the ICE lines are colored
+    palette: str
+        Name of the matplotlib color palette
+
+    Returns
+    -------
+    list of colors
+    norm: mpl.colors.Normalize
+    """
+    cmap = mpl.cm.get_cmap(palette)
+    norm = mpl.colors.Normalize(vmin=value_list.min(), vmax=value_list.max())
+    color_list = [mpl.colors.to_hex(cmap(norm(value))) for value in value_list]
+    return color_list, norm
+
+
+def list_of_values_passed_for_ice_line_colors(ice_lines_kw=None):
+    """
+    Checks if the user passed a list of values by which the individual ice lines
+    should be colored
+
+    Parameters
+    ----------
+    ice_lines_kw : dict, default=None
+            Dictionary with keywords passed to the `matplotlib.pyplot.plot` call.
+            For ICE lines in the one-way partial dependence plots.
+
+    Returns
+    -------
+    True if list of values is passed, False else
+    """
+    # case 1: no color is given by the user
+    if ice_lines_kw is None:
+        return False
+    elif "color" not in ice_lines_kw.keys():
+        return False
+    # case 2: Tuple as color is not valid, but will throw matplotlib error later on
+    elif isinstance(ice_lines_kw["color"], tuple):
+        return False
+    # case 3: color argument can be a single color for all lines
+    elif np.array(ice_lines_kw["color"]).size == 1:
+        return False
+    # case 4: list of values is passed
+    else:
+        return True
+
+
+def get_color_list_and_legend_dict(ice_lines_kw):
```

Comment:
da du nur 2 werte aus dem dict nimmst, würde ich die stattdessen direkt übergeben, dann kannst du auch da den default wert für `palette` setzen

Reply:
Dann bräuchte ich ja wieder einen Check in add_legend_or_cmap_for_individually_colored_ice_lines, ob das categorical or continuous ist. 
Aktuell ist es so:
`if "mapping" in legend_dict.keys():`
`elif "norm" in legend_dict.keys():`